setuptools/__init__.py

```diff
@@ -108,11 +108,12 @@ def _fetch_build_eggs(dist: Distribution):
         raise
 
 
-def setup(**attrs):
+def setup(**attrs) -> Distribution:
     logging.configure()
     # Make sure we have any requirements needed to interpret 'attrs'.
     _install_setup_requires(attrs)
-    return distutils.core.setup(**attrs)
+    # Override return type of distutils.core.Distribution with setuptools.dist.Distribution
```

Comment:
```suggestion
    # Override return type of distutils.core.Distribution with setuptools.dist.Distribution
    # (implicitly implemented via `setuptools.monkey.patch_all`).
```

This is just an optional clarification note. Not necessarily you need to act on that, only if you are planning to change the comments/ignores in `develop` (I can add it locally later, ).

Reply:
In principle, we could modify those comments for clarity but possibly not in this PR?

I think there is also a subtle difference between the definition inside of a `Command` class and inside the `setup()` function:

For the command classes, the `Distribution` object is passed in as an [argument to the constructor](https://github.com/pypa/setuptools/blob/main/setuptools/_distutils/cmd.py#L89), so there is no direct patching of type of the given object `Distribution` itself[^1]. It is a "dependency injection" kind of situation, the object simply preserves the same type it had when it was passed in. Possibly that could be co-variant but I guess it would be impossible to satisfy static analysis given that `distutils.core.Distribution` is patched somewehere else in runtime?

On the other hand, inside `setup(...)` we are directly forwarding the call to `distutils.core.setup(...)`, so I think that the comment helps to understand the "misterious runtime behaviour", when it is expected that `distutils.core.setup(...)` should return the `distutils` version of `Distribution` (but in runtime it does not).

[^1]: Despite `distutils.cmd.Distribution` being patched, note how that [module](https://github.com/pypa/setuptools/blob/main/setuptools/_distutils/cmd.py) only uses `Distribution` for dynamic checks but never instantiate an object.